statsmodels/discrete/tests/test_discrete.py

```diff
@@ -1646,8 +1646,7 @@ def test_negative_binomial_default_alpha_param():
     with warnings.catch_warnings():
         warnings.simplefilter("error")
         sm.families.NegativeBinomial(alpha=1.0)
-    with warnings.catch_warnings():
-        warnings.simplefilter("error")
```

Comment:
The warning for negbin is a UserWarning, AFAIR
so this could be more selective to handle both kinds of warning

I guess the following will only raise the UserWarning

sm.families.NegativeBinomial(link=sm.families.links.NegativeBinomial(alpha=1.0),
                                     alpha=1.0)